lstchain/reco/disp.py

```diff
@@ -36,7 +37,16 @@ def disp(cog_x, cog_y, src_x, src_y):
     """
     disp_dx = src_x - cog_x
     disp_dy = src_y - cog_y
-    disp_norm = np.sqrt(disp_dx**2 + disp_dy**2)
+    
+    disp_norm = disp_dx * np.cos(hillas_psi) + disp_dy * np.sin(hillas_psi)
+    disp_sign = np.sign(disp_norm)
+    disp_norm = abs(disp_norm)
```

Comment:
Since this is changing the definition of `disp_norm` and `disp_sign`, shouldn't the reconstruction should be changed accordingly? (I mean applying the reconstructed disp to get the source position)

Reply:
What do you want to change? If anything, this new definition brings it more in line with how it is applied. Using the distance along the shower axis, not the full 2d distance.

Reply:
No, why? The definition of disp_sign, in what refers to how it is used, is the same (though before its true value was not correctly calculated). For disp_norm there is indeed a small change, it is now the cog-to-source distance projected on the reconstructed image axis. But that is (conceptually) more correct than using directly the distance. In practice it won't change much for gammas, the two things agree for Alpha=0 and are nearly identical for all relevant values of Alpha.

Reply:
>  But that is (conceptually) more correct than using directly the distance. 

Ha yes, you are right.